src/MessageList/MessageListItem.vala

```diff
@@ -103,19 +103,43 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
 
         settings = new GLib.Settings ("io.elementary.mail");
 
-        web_view = new Mail.WebView ();
-        web_view.margin = 6;
-        web_view.image_load_blocked.connect (() => {
-            // TODO: Show infobar
+        var infobar = new Gtk.InfoBar ();
+        infobar.margin = 12;
+        infobar.message_type = Gtk.MessageType.WARNING;
+        infobar.add_button (_("Show Images"), 1);
+        infobar.add_button (_("Always Show from Sender"), 2);
+        infobar.get_style_context ().add_class (Gtk.STYLE_CLASS_FRAME);
+        infobar.no_show_all = true;
+
+        var infobar_content = infobar.get_content_area ();
+        infobar_content.add (new Gtk.Label (_("This message contains remote images.")));
+        infobar_content.show_all ();
+
+        ((Gtk.Box) infobar.get_action_area ()).orientation = Gtk.Orientation.VERTICAL;
+
+        infobar.response.connect ((id) => {
+            var sender = message_info.from;
```

Comment:
re-adding a note that this needs to get just the email address without the sender's name